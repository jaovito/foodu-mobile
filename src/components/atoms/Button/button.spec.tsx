import {render, fireEvent} from '@test-utils';
import {Button} from '.';

describe('Button Tests', () => {
  const text = 'Hello world!';

  it('should render correctly', () => {
    const button = render(<Button>{text}</Button>);

    expect(button).toBeDefined();
  });

  it('should contains the existing text', async () => {
    const {findByText} = render(<Button>{text}</Button>);

    const button = await findByText(text);

    expect(button).toBeTruthy();
  });

  it('should be able to click on the button', async () => {
    const onPress = jest.fn();

    const {getByTestId} = render(<Button onPress={onPress}>{text}</Button>);

    fireEvent.press(getByTestId('ButtonTestID'));

    expect(onPress).toHaveBeenCalled();
  });

  it('should be able to click 2 times on the button', async () => {
    const onPress = jest.fn();

    const {getByTestId} = render(<Button onPress={onPress}>{text}</Button>);

    fireEvent.press(getByTestId('ButtonTestID'));
    fireEvent.press(getByTestId('ButtonTestID'));

    expect(onPress).toHaveBeenCalledTimes(2);
  });

  it('should match with snapshot (with on press)', async () => {
    const onPress = jest.fn();

    const button = render(<Button onPress={onPress}>{text}</Button>).toJSON();

    expect(button).toMatchSnapshot();
  });

  it('should match with snapshot (without onPress)', async () => {
    const button = render(<Button>{text}</Button>).toJSON();

    expect(button).toMatchSnapshot();
  });
});
