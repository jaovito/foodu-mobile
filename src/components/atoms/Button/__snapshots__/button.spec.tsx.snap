// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Tests should match with snapshot (with on press) 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FE8C00",
      "borderRadius": 16,
      "justifyContent": "center",
      "marginTop": 16,
      "opacity": 1,
      "paddingBottom": 31,
      "paddingLeft": 31,
      "paddingRight": 31,
      "paddingTop": 31,
      "width": "100%",
    }
  }
  testID="ButtonTestID"
>
  <Text
    style={
      [
        {
          "color": "#0D0D0D",
          "fontFamily": "Poppins_400Regular",
          "fontSize": 31,
        },
        {
          "color": "#FFFFFF",
        },
      ]
    }
  >
    Hello world!
  </Text>
</View>
`;

exports[`Button Tests should match with snapshot (without onPress) 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FE8C00",
      "borderRadius": 16,
      "justifyContent": "center",
      "marginTop": 16,
      "opacity": 1,
      "paddingBottom": 31,
      "paddingLeft": 31,
      "paddingRight": 31,
      "paddingTop": 31,
      "width": "100%",
    }
  }
  testID="ButtonTestID"
>
  <Text
    style={
      [
        {
          "color": "#0D0D0D",
          "fontFamily": "Poppins_400Regular",
          "fontSize": 31,
        },
        {
          "color": "#FFFFFF",
        },
      ]
    }
  >
    Hello world!
  </Text>
</View>
`;
